tokio-trace/src/level_filters.rs

```diff
@@ -0,0 +1,121 @@
+use std::cmp::Ordering;
+use tokio_trace_core::Level;
+
+/// Describes the level of verbosity of a span or event.
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+pub struct LevelFilter(LevelFilterInner, Option<Level>);
```

Comment:
Adding an `Option<Level>` to the `LevelFilter` type is an interesting approach I hadn't actually thought of. I would have just had a separate `LevelFilterInner` type and implemented the comparison with `Level` via a match.  Your solution seems more elegant to me --- good thinking!

With that said, I think the `LevelFilterInner` type shouldn't actually be necessary --- a `LevelFilter` should be representable as just
```suggestion
pub struct LevelFilter(Option<Level>);
```